heed-traits/src/lib.rs

```diff
@@ -11,3 +12,44 @@ pub trait BytesDecode<'a> {
 
     fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem>;
 }
+
+impl<'a, A, B> BytesEncode<'a> for (A, B)
+where
+    A: BytesEncode<'a>,
+    A::EItem: Sized,
+    B: BytesEncode<'a>,
+{
+    type EItem = (A::EItem, B::EItem);
+
+    fn bytes_encode(item: &'a Self::EItem) -> Option<Cow<[u8]>> {
+        let first_bytes = A::bytes_encode(&item.0)?;
+        let second_bytes = B::bytes_encode(&item.1)?;
+        let mut buffer = Vec::with_capacity(
+            mem::size_of::<usize>()
+            + &first_bytes.len()
+            + &second_bytes.len());
+        // signify the length of the first item
+        buffer.extend_from_slice(&first_bytes.len().to_be_bytes());
+        buffer.extend_from_slice(&first_bytes);
+        buffer.extend_from_slice(&second_bytes);
+        Some(Cow::Owned(buffer))
+    }
+}
+
+impl<'a, A, B> BytesDecode<'a> for (A, B)
+where
+    A: BytesDecode<'a>,
+    B: BytesDecode<'a>,
+{
+    type DItem = (A::DItem, B::DItem);
+
+    fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem> {
+        let mut size_buf = [0; 8];
+        size_buf.copy_from_slice(bytes.get(0..8)?);
```

Comment:
You used `size_of<usize>` above but here you use 8, hum....

Reply:
good point, gonna fix this